### PR TITLE
Feat html rewriting

### PR DESCRIPTION
**Pull request by Folyd**
This is a successor of #140.

### CODE REVIEW

File: src/html.rs [hunk 1]
```diff
@@ -79,27 +87,40 @@ pub fn rewrite_html_base_url(
...
-                        if attr.starts_with("background-image: url('/") {
-                            let mut base_url = "";
-                            if let Some(url) = site_url {
-                                base_url = url;
+                    if let Some(style) = el.get_attribute("style") {
+                        let mut pairs = helpers::split_styles(&style);
+                        let backgrond_image_url = match pairs.get("background-image") {
+                            Some(value) if CSS_STATIC_URL_REGEX.is_match(value) => {
+                                if let Some(cdn_url) = cdn_url {
+                                    value.replacen("/static", cdn_url, 1)
```
Comment (l1ch40) [inline src/html.rs:95]: the replacen isn't an in-place modification, therefore it should be assigned the value

File: src/html.rs [hunk 2]
```diff
@@ -79,27 +87,40 @@ pub fn rewrite_html_base_url(
...
+                                    return Ok(());
+                                }
                             }
-                            if attr.starts_with("background-image: url('/static") {
-                                attr = attr.replacen("/static", "", 1);
-                                if let Some(url) = cdn_url {
-                                    base_url = url;
+                            Some(value) if CSS_ROOT_URL_REGEX.is_match(value) => {
+                                if let Some(site_url) = site_url {
+                                    value.replacen('/', &format!("{site_url}/"), 1)
```
Comment (l1ch40) [inline src/html.rs:102]: the replacen isn't an in-place modification, therefore it should be assigned the value

File: src/html.rs [hunk 3]
```diff
@@ -9,6 +11,20 @@ use markup5ever_rcdom::{Handle, NodeData, RcDom};
...
 
+use crate::helpers;
+
+// Generated by ChatGPT.
+//
+// The regular expression can be used to match the CSS image url:
+//
+// (?i) - This is an "inline" regular expression option that makes the regular expression case-insensitive. This means that it will match "url(" as well as "URL(".
+// url\( - Matches the characters "url(" literally (case-insensitively, due to the (?i) option)
+// '? - Matches either a single quote, or none at all.
```
[the remark this answers is not in the document]
Comment (Folyd) [reply src/html.rs:22]: The main reason is `lol_html` doesn't support double quotes in `url()` function, I tested it the unit test with double quotes will be failed. 
```
---- html::tests::test_rewrite_background_image_url::lowercase stdout ----
background-image: url(
[src/html.rs:96] helpers::split_styles(&style) = {
    "background-image": "url(",
}
[src/html.rs:97] pairs.get("background-image") = Some(
    "url(",
)
[src/html.rs:105] CSS_ROOT_URL_REGEX.is_match(value) = false
thread 'html::tests::test_rewrite_background_image_url::lowercase' panicked at 'assertion failed: `(left == right)`
  left: `"<body><div class=\"bg-primary text-main\" style=\"background-image: url(\"/test.png\");\"></div></body>"`,
 right: `"<body><div class=\"bg-primary text-main\" style=\"background-image: url(\"https://github.com/test.png\");\"></div></body>"`', src/html.rs:273:9
```

File: src/html.rs (hunk 1 above)
Comment (Folyd) [reply src/html.rs:95]: According to the docs: `replacen creates a new String, and copies the data from this string slice into it.`, I think it works as expected.

File: src/html.rs (hunk 2 above)
Comment (Folyd) [reply src/html.rs:102]: Same as above

File: src/html.rs (hunk 1 above)
Comment (l1ch40) [reply src/html.rs:95]: seems wrong: 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f2912d0d07994a1d45915a1bf677fe47

File: src/html.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (l1ch40) [reply src/html.rs:22]: > The main reason is `lol_html` doesn't support double quotes in `url()` function, I tested it the unit test with double quotes will be failed.
> 
> ```
> ---- html::tests::test_rewrite_background_image_url::lowercase stdout ----
> background-image: url(
> [src/html.rs:96] helpers::split_styles(&style) = {
>     "background-image": "url(",
> }
> [src/html.rs:97] pairs.get("background-image") = Some(
>     "url(",
> )
> [src/html.rs:105] CSS_ROOT_URL_REGEX.is_match(value) = false
> thread 'html::tests::test_rewrite_background_image_url::lowercase' panicked at 'assertion failed: `(left == right)`
>   left: `"<body><div class=\"bg-primary text-main\" style=\"background-image: url(\"/test.png\");\"></div></body>"`,
>  right: `"<body><div class=\"bg-primary text-main\" style=\"background-image: url(\"https://github.com/test.png\");\"></div></body>"`', src/html.rs:273:9
> ```

Get, Can we process it after replace double-quote instead to single-quote ?

**Comment by l1ch40** [comment]
LGTM

**Pull request merged**